cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -501,6 +505,110 @@ std::vector<size_type> segmented_null_count(bitmask_type const* bitmask,
   return detail::segmented_count_unset_bits(bitmask, indices_begin, indices_end, stream);
 }
 
+/**
+ * @brief Reduce an input null mask using segments defined by offset indices
+ * into an output null mask.
```

Comment:
Reduced how? What does it mean to "reduce" a null mask?

Reply:
It's arguably "reduction" is a bit too broad for this operation. But it might be easier for user of the function to understand the net outcome of the function as a fold with binary `AND` of all the bits if `null_handling` is `INCLUDE`, `OR` otherwise. (Though implementation doesn't really do that).

Reply:
Beyond the `AND`/`OR`, there's special behavior for handling empty segments. This is hard to succinctly describe, and perhaps a generic `@brief` followed by a more extensive description is the best we can do. I've tried to come up with a better way to write this but haven't come up with anything I like.

Reply:
Maybe:
```suggestion
 * @brief Create an output null mask whose validity is determined by the
 * validity of any/all elements of segments of an input null mask.
```